### PR TITLE
Add AttendanceEventSubmission (3/?)

### PR DESCRIPTION
**Pull request by jessouyangbraven**
### Description

**Task**: https://app.asana.com/0/1174274412967132/1197571375872706

This PR adds the DB migrations, models, and specs for `AttendanceEventSubmission` and `AttendanceEventSubmissionAnswer`.

Follow-up PRs will implement:
- `AttendanceEventSubmissionAnswersController#launch`: what the Fellow sees when they click on the `CourseAttendanceEvent` assignment in the Accelerator course.
- `AttendanceEventSubmissionsController#launch`: what the LC sees when they click on **Take Attendance** in the LC Playbook course. This is where we will show a drop-down of all the `CourseAttendanceEvent`s for the corresponding Accelerator course as well as a form to take attendance for that event.

### Motivation and Context

The `Submission`s and `SubmissionAnswer`s are created by the LC when they take attendance in the LC Playbook course. 

There is one `Submission` per LC for a `CourseAttendanceEvent`. Each `Answer` belongs to a `Submission` and is the attendance status of the Fellow specified in `for_user` for that particular `CourseAttendanceEvent`.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/models/attendance_event_submission.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  belongs_to :course_attendance_event
+
+  has_one :course, through: :course_attendance_event
+
+  validates :user, :course_attendance_event, presence: true
+
+  def save_answers!(answers_params_hash)
+    # TODO: In the next PR, when we add the equivalent of SubmissionAnswers
+    # model for this submission type
+  end
```
Comment (jessouyangbraven) [inline app/models/attendance_event_submission.rb:14]: RFC: @sadleb brought this up when I implemented `FellowEvaluationSubmission`. At the time, I was pattern matching on existing submissions and not thinking ahead to this attendance stuff.

Should we have an `AttendanceEventSubmission` per fellow being reviewed by the LC, or stay consistent with existing submission data models and have `AttendanceEventSubmissionAnswer` (or `Status` or whatever makes sense to call it), one for each user, that all `belong_to` the `Submission`?

There are two categories of `Submissions`:
1. Submissions that involve you filling out the same fields for a list of `target_user`s: `{PeerReview, FellowEvaluation, AttendanceEvent}Submissions` fall into this category

2. Submissions that involve you filling out a bunch of a stuff for a single logical "form", `{Waivers, PreAcceleratorSurvey, PostAcceleratorySurvey, Project, Survey}Submission` fall into this category.

Comment (jessouyangbraven) [reply app/models/attendance_event_submission.rb:14]: We don't store anything for waivers, pre/post accelerator surveys. So I think the question I'm trying to ask if whether the data model for category (1) should be different for `Project` and `SurveySubmissions`.

Comment (jessouyangbraven) [reply app/models/attendance_event_submission.rb:14]: Screen Shot 2021-01-14 at 3 32 39 PM

Comment (jessouyangbraven) [reply app/models/attendance_event_submission.rb:14]: Screen Shot 2021-01-14 at 3 32 25 PM

Comment (jessouyangbraven) [reply app/models/attendance_event_submission.rb:14]: https://app.asana.com/0/inbox/1168520191375327/1199544270990456/1199539421634484

Comment (jessouyangbraven) [reply app/models/attendance_event_submission.rb:14]: https://app.asana.com/0/1174274412967132/1199544270990456

Comment (jessouyangbraven) [reply app/models/attendance_event_submission.rb]: We went with keeping things consistent with existing `Submission` and `Answer` models, so having the `for_user_id` on the `submission_answers` table.

**Review by rshipp — APPROVED** [review]

File: db/migrate/20210114220410_create_attendance_event_submissions.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class CreateAttendanceEventSubmissions < ActiveRecord::Migration[6.0]
+  def change
+    create_table :attendance_event_submissions do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :course_attendance_event, null: false, foreign_key: true, index: { name: :index_submissions_on_course_attendance_event_id }
+    end
```
Comment (sadleb) [inline db/migrate/20210114220410_create_attendance_event_submissions.rb:6]: We're missing the timestamps for `created_at` and `updated_at`

**Review by sadleb — CHANGES_REQUESTED** [review]

**Review by sadleb — APPROVED** [review]
lgtm

**Pull request merged**